### DISCUSSION TITLE
What license do the icons for widget.CurrentLayoutIcon have?

### DISCUSSION THREAD

**Discussion by thingsiplay** [qtile/qtile#3490] [General]
After my spectacular fail and at least a good outcome at https://github.com/qtile/qtile/issues/3489 (thanks to the devs!), I wrote a few scripts to generate different colors from the base Icons for widget.CurrentLayoutIcon. And I plan on making brand new set (I am not an artist, so nothing special), but who knows how long this can take.

My question is, what license do the icons have? Does ~GPL~ (Edit: *oopsie, it is __MIT License__, not GPL*) apply to the images? Because I am thinking of generating and distributing bunch of colored variations of the original icons.

**Comment by m-col** [comment]
They are licensed the same as the code, which is MIT

**Comment by thingsiplay** [comment]
Ooops it's MIT for the code. Today I make a lot of oopsies... Thanks for the quick reply.

**Reply by m-col** [reply]
No probs. Coloured variations sound cool. I wonder if it would be possible to have them as SVGs in a way where the colour can be configured at runtime.

**Reply by thingsiplay** [reply]
Qtile can generate the images from SVGs anyway, right? So this would be an obvious feature. And would be much cooler. I am currently just doing it with ImageMagick on the .png with Bash. The icons was the only white color on my bar and it bugged me since last year.

**Reply by m-col** [reply]
It can render SVGs on the bar, if that's what you mean. But it cannot generate SVGs. As PNG is a raster image format, I don't think you can convert to SVG and have the lines be configurable colours, as the icons would be embedded in the SVG as a raster (not 100% certain). You'd need to draw them again from scratch in inkscape or something, with paths. However I don't know how you could change the colour at runtime.

**Reply by thingsiplay** [reply]
The PNGs can easily converted to vector format with Inkscape. It has trace feature and I used it couple of times for other things. https://i.imgur.com/7rLa4fi.png
It depends on what Qtile can do with the SVGs. If not, a script could be used to convert them to PNG on demand by activating specific combinations of layers or similar. SVG files are just XML anyway.

**Reply by m-col** [reply]
Huh, cool! I use inkscape every now and then but never came across this feature, pretty nice :smile: 

This code: https://github.com/qtile/qtile/blob/master/libqtile/widget/statusnotifier.py#L182-L185 suggests that SVGs should work. However `libqtile.images` doesn't look like it supports them. @elParaguayo would you be able to shed some light on this?

**Reply by elParaguayo** [reply]
Yes, SVG images are supported (I had some code somewhere which shows which image types are supported, I'll see if I can dig it up later).

In terms of changing the colour, I've actually done something like that in my qtile-extras repo. There's an `ImgMask` class which uses the source image as a mask and you just paint any colour you want. https://qtile-extras.readthedocs.io/en/latest/manual/how_to/img-mask.html#img-mask

I don't know if it's feasible to merge this code into the main repo (we're not adding too many new features). However, it would be very easy to create a version of the widget using the mask. If be happy to include that in my repo.

**Reply by elParaguayo** [reply]
Using `cairocffi`, the following image formats can be imported into an `ImageSurface`: `ani`, `png`, `bmp`, `gif`, `ico`, `jpeg`, `pnm`, `tiff`, `xpm`, `xbm`, `tga`, `icns`, `qtif`, `wmf`, `avif`, `heif/avif`, `jxl`, `svg`
